mmedit/models/editors/pggan/pggan.py

```diff
@@ -161,13 +161,15 @@ def forward(self,
 
         outputs = _model(
             noise, curr_scale=curr_scale, transition_weight=transition_weight)
+        outputs = self.data_preprocessor.destruct(outputs, data_samples)
 
         if sample_model == 'ema/orig':
             _model = self.generator
             outputs_orig = _model(
                 noise,
                 curr_scale=curr_scale,
                 transition_weight=transition_weight)
+            outputs = self.data_preprocessor.destruct(outputs, data_samples)
             outputs = dict(ema=outputs, orig=outputs_orig)
```

Comment:
here, `outputs` is constructed as a dictionary, and in the following codes, we fetch the values and pass them to `gen_sample`. So why not directly pass the values to the `gen_sample`. Besides, why naming the data_sample as `gen_sample`?

Reply:
check these points for all the models.